Some ideas for changes to the incident playbook (#63)
DFE-Digital/teacher-services-tech-docs: source/operating-a-service/incident-playbook.html.md

1.  the **Incident comms lead**
    (responsible for communicating and alerts to users and stakeholders)
2.  the **Incident tech lead**
    (responsible for technical direction for fixing the incident and communicating to the comms lead)
3. the **Incident support lead**
    (responsible for monitoring Zendesk and alerting the team to any changes in the queues and severity of experience of users).

Who will usually be in these lead roles
- The comms lead will typically be the delivery manager of the main affected service. If that person is not available, the product manager could take on this role, or a DM from another service.
- The tech lead will typically be the tech lead of the main affected service, or another developer from the team.
- The support lead will typically be assigned by the support team.

> For an incident that is cross-service, try to appoint whoever is available. If an incident is confirmed, it is best to declare the incident as soon as possible. You can always update who is in these roles later.

Reviewer: ```suggestion
For an incident that is cross-service, try to appoint whoever is available; the comms lead would usually be the program delivery manager or an experienced DM from one of the service teams, and the tech lead would usually be the lead developer or a TL from one of the service teams. If an incident is confirmed, it is best to declare the incident as soon as possible. You can always update who is in these roles later.
```

### 2. Triage the incident (any incident lead)

The person declaring the incident should attempt to [triage the incident](/operating-a-service/how-to-categorise-technical-incidents.html) (P1, P2, P3).

You can always update this later if needed.

### 3. Create an incident Slack channel (any incident lead)

Initiate the Slack IncidentBot by sending the message `/incident open` in your service Slack channel or #teacher-services-infra. 

> Complete the details in the IncidentBot template, and press Enter, which will automatically create a dedicated Slack channel for the incident.

Author: Good point. I don't think adding everyone should be done here, but any leads and others vital to solving the incident is important. The comms lead can then focus on communicating to others whilst tech folk try to fix it?

> Complete the details in the IncidentBot template, and press Enter, which will automatically create a dedicated Slack channel for the incident.

Author: ```suggestion
Complete the details in the IncidentBot template, and press Enter, which will automatically create a dedicated Slack channel for the incident.

Add vital people as soon as you can to the Slack channel. The comms lead can focus on adding any others later.
```

> Complete the details in the IncidentBot template, and press Enter, which will automatically create a dedicated Slack channel for the incident.

Author: amended

### 4. Focus on fixing the incident (usually tech lead)

The tech lead should be focused on either fixing the incident themselves or giving direction to other developers to do so.

> If the incident is not of a technichal nature, for example, as the result of a data breach, the focus should still be on resolving the incident. 

Reviewer: Not sure I really get the significance of this sentence.

> If the incident is not of a technichal nature, for example, as the result of a data breach, the focus should still be on resolving the incident. 

Author: ```suggestion
If the incident is not of a technical nature, for example, as the result of a data breach, others might lead on resolving the incident.
```

### 5. Determine who to contact and how (comms lead)

Determine who needs to be contacted, based on the incident priority and affected services. 

Use the contacts from the [Teacher services list](https://educationgovuk.sharepoint.com.mcas.ms/sites/teacher-services-infrastructure/Lists/Teacher%20services%20list/AllItems.aspx) and your service's contact list if you have one.

It may be best to include critical user groups like lead providers, or other external organisations we work with. Make sure to include programme delivery managers, deputy directors and service owners in case of a P1 incident.

You need to decide how to contact them as well. You might need to use email, Teams, Slack or a combination to make sure you reach everyone. You can also add them to the Slack incident channel. 

### 6. Identify any other services involved (any leads)

The leads should identify:

- any upstream services (both inside and outside DfE) which could be contributing to the issue

> - any downstream services (both inside and outside DfE) likely to be affected by the issue and raise incidents where needed

Reviewer 2: We may need to review the criticality is other services are impacted

Author: agree, amended

> - any downstream services (both inside and outside DfE) likely to be affected by the issue and raise incidents where needed

Author: ```suggestion
- any downstream services (both inside and outside DfE) likely to be affected by the issue
```

> - any downstream services (both inside and outside DfE) likely to be affected by the issue and raise incidents where needed
> 
> After doing this, you might realise you need to contact those teams as well. This is the responsibility of the comms lead.

Reviewer: ```suggestion
- any downstream services (both inside and outside DfE) likely to be affected by the issue

It may be necessary to contact the upstream/downstream service teams and raise incidents in those services. This is the responsibility of the comms lead.
```

> After doing this, you might realise you need to contact those teams as well. This is the responsibility of the comms lead.

Author: ```suggestion
After doing this, you might review the priority of the incident. You might also need to contact those teams as well. This is the responsibility of the comms lead.
```

Informing users about incidents is generally considered best practice, but should be decided on a case by case basis with the product and service managers.

> The Teacher Services team maintains a publicly available [service status dashboard](https://teacher-services-status.education.gov.uk/). For some incidents, you might find it helpful to update this. You will need a GitHub account to do this, or delegate updates to a colleague who has one.

Reviewer: ```suggestion
The Teacher Services team maintains a publicly available [service status dashboard](https://teacher-services-status.education.gov.uk/), generated by our [Upptime](https://github.com/DFE-Digital/teacher-services-upptime) instance on GitHub.
```

Reviewer: The updates are automatic

Author: Oh wow! it didn't say that before - said the comms lead needed to update it!

The updates are managed via GitHub Actions and issues on the [teacher-services-upptime repository on GitHub](https://github.com/DFE-Digital/teacher-services-upptime). If a service’s automatic health check is failing continuously, an issue will be created within 5 minutes of the failure occurring and the dashboard will start reporting a service issue.

To update the dashboard:

1. Navigate to the appropriate incident issue on the [GitHub issues page](https://github.com/DFE-Digital/teacher-services-upptime/issues)

> 2. Add a comment to the issue

Reviewer: This is quite a big change. Do we want this? In my experience, consolidating discussion in the Slack incident channel works pretty well and is delightfully accessible.

### 9. If the incident requires invocation of Disaster Recovery procedures

> Follow the [Disaster Recovery procedure](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/disaster-recovery.md)

Reviewer 2: and any specific documentation in the service

> Follow the [Disaster Recovery procedure](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/disaster-recovery.md)

Author: ```suggestion
Follow the [Disaster Recovery procedure](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/disaster-recovery.md) and any other specific documentation for the service.
```

## Provide updates during the incident (usually comms lead)

Keep all conversations and status updates about the incident on the dedicated Slack incident channel.

Use the Slack IncidentBot `/incident update` command to update:

- Description
- Priority
- Leads

> Update stakeholders on the Slack incident channel everytime something changes or everyday, until the incident has been resolved. 

Author: ```suggestion
Update stakeholders on the Slack incident channel regularly, until the incident has been resolved. This might be every hour for high risk incidents, but at least every time something changes. Make sure your team is kept updated too by linking them to the incident channel.
```

Reviewer 3: looks good
